packages/app/src/cli/services/init/validate.ts

```diff
@@ -0,0 +1,63 @@
+import {isPredefinedTemplate, templates, visibleTemplates} from '../../prompts/init/init.js'
+import {safeParseURL} from '@shopify/cli-kit/common/url'
+import {AbortError} from '@shopify/cli-kit/node/error'
+import {outputContent, outputToken} from '@shopify/cli-kit/node/output'
+
+export function validateTemplateValue(template: string | undefined) {
+  if (!template) {
+    return
+  }
+
+  const url = safeParseURL(template)
+  if (url && url.origin !== 'https://github.com')
```

Comment:
I don't remember, but the code that downloads the template assumes is a github repo everywhere 🤔 Maybe because we wanted to support a specific URL format with branches and everything.